src/main/java/com/aws/greengrass/clientdevices/auth/session/SessionImpl.java

```diff
@@ -45,4 +45,24 @@ public DeviceAttribute getSessionAttribute(String attributeNamespace, String att
         }
         return null;
     }
+
+    /**
+     * Check if attribute exists.
+     *
+     * @param attributeNameSpace Attribute namespace
+     * @param attributeName      Attribute name
+     * @return True/false
+     */
+    @Override
+    public boolean containsSessionAttribute(String attributeNameSpace, String attributeName) {
+
+        if (!this.containsKey(attributeNameSpace)) {
```

Comment:
there's still a race between check and get here, as well as in `updateResource`.  i think we can remove the contains method and just use `getSessionAttribute` in `updateResource`